config/src/test/java/io/confluent/common/config/ConfigDefTest.java

```diff
@@ -275,4 +285,79 @@ private void testValidators(Type type, ConfigDef.Validator validator, Object def
       }
     }
   }
+
+  @Test
+  public void testClassLoading() throws IOException {
+    ClassLoader systemLoader = ClassLoader.getSystemClassLoader();
+
+    // test with system classloader
+    String configName = "type.class.property";
+    String configNameDefault = Password.class.getName();
+
+    ConfigDef configDef = new ConfigDef();
+    configDef.define(configName, Type.CLASS, configNameDefault,
+        ConfigDef.Importance.MEDIUM, "system class");
+
+    AbstractConfig config = new AbstractConfig(configDef, new HashMap<>());
+
+    Class<?> systemKlass = config.getClass(configName);
+    assertEquals(systemLoader, systemKlass.getClassLoader());
+
+    // test with custom, child-first, classloader
+    Set<String> allowedKlasses = Collections.singleton(configNameDefault);
+    List<URL> pluginUrls = Collections.singletonList(Paths.get("target/classes/").toUri().toURL());
+    URLClassLoader pluginLoader = new PluginClassLoader(pluginUrls.toArray(new URL[0]), systemLoader, allowedKlasses);
+
+    Thread.currentThread().setContextClassLoader(pluginLoader);
+
+    configDef = new ConfigDef();
+    configDef.define(configName, Type.CLASS, configNameDefault, ConfigDef.Importance.MEDIUM, "plugin class");
+
+    config = new AbstractConfig(configDef, new HashMap<>());
```

Comment:
Would it make sense to go a bit further and try getting the class from `config.getClass(configName)`, both here and above?

Reply:
Not sure I follow the suggestion. 

`config.getClass(configName)` is called indeed twice. 
Once in line 303 and once in 318 (current line numbers). 

If you are referring to `configNameDefault = Password.class.getName();` probably better to not call this to leave it to the ConfigDef parsing to load this class when it constructs the defintion of the config. (Doesn't make a difference if we add this call before the definition).

Reply:
I missed that. Apologies.